test/gtest/uct/test_md.cc

```diff
@@ -751,68 +770,87 @@ UCS_TEST_SKIP_COND_P(test_md, dereg_bad_arg,
-    params.memh        = memh;
-    params.flags       = UCT_MD_MEM_DEREG_FLAG_INVALIDATE;
-    params.comp        = &comp().comp;
-
-    if (!check_caps(UCT_MD_FLAG_INVALIDATE)) {
-        params.field_mask = UCT_MD_MEM_DEREG_FIELD_COMPLETION |
-                            UCT_MD_MEM_DEREG_FIELD_FLAGS |
-                            UCT_MD_MEM_DEREG_FIELD_MEMH;
-        status            = uct_md_mem_dereg_v2(md(), &params);
-        ASSERT_UCS_STATUS_EQ(UCS_ERR_UNSUPPORTED, status);
-
-        params.field_mask = UCT_MD_MEM_DEREG_FIELD_MEMH;
-        status            = uct_md_mem_dereg_v2(md(), &params);
-    } else {
-        params.field_mask = UCT_MD_MEM_DEREG_FIELD_COMPLETION;
-        status            = uct_md_mem_dereg_v2(md(), &params);
-        ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
-
-        params.field_mask = UCT_MD_MEM_DEREG_FIELD_COMPLETION |
-                            UCT_MD_MEM_DEREG_FIELD_FLAGS;
-        status            = uct_md_mem_dereg_v2(md(), &params);
-        ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
-
-        params.field_mask = UCT_MD_MEM_DEREG_FIELD_MEMH |
-                            UCT_MD_MEM_DEREG_FIELD_FLAGS;
-        status            = uct_md_mem_dereg_v2(md(), &params);
-        ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
-
-        params.field_mask = UCT_MD_MEM_DEREG_FIELD_MEMH |
-                            UCT_MD_MEM_DEREG_FIELD_COMPLETION |
-                            UCT_MD_MEM_DEREG_FIELD_FLAGS;
-        status            = uct_md_mem_dereg_v2(md(), &params);
-        ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
-
-        std::vector<uint8_t> rkey(md_attr().rkey_packed_size);
-        uct_md_mkey_pack_params_t pack_params;
-        pack_params.field_mask = UCT_MD_MKEY_PACK_FIELD_FLAGS;
-        pack_params.flags      = UCT_MD_MKEY_PACK_FLAG_INVALIDATE;
-        status = uct_md_mkey_pack_v2(md(), memh, &pack_params, rkey.data());
-        EXPECT_UCS_OK(status);
+        ptr    = malloc(size);
+        status = reg_mem(md_flags, ptr, size, &memh);
+        ASSERT_UCS_OK(status);
 
-        status = uct_md_mem_dereg_v2(md(), &params);
-    }
+        comp().comp.func   = dereg_cb;
+        comp().comp.count  = 1;
+        comp().comp.status = UCS_OK;
+        comp().self        = this;
+        params.memh        = memh;
+        params.flags       = UCT_MD_MEM_DEREG_FLAG_INVALIDATE;
+        params.comp        = &comp().comp;
+
+        if (is_unsupported_caps(md_flags)) {
+            params.field_mask = UCT_MD_MEM_DEREG_FIELD_COMPLETION |
+                                UCT_MD_MEM_DEREG_FIELD_FLAGS |
+                                UCT_MD_MEM_DEREG_FIELD_MEMH;
+            status            = uct_md_mem_dereg_v2(md(), &params);
+            ASSERT_UCS_STATUS_EQ(UCS_ERR_UNSUPPORTED, status);
+
+            params.field_mask = UCT_MD_MEM_DEREG_FIELD_MEMH;
+            status            = uct_md_mem_dereg_v2(md(), &params);
```

Comment:
isn't this code duplication of some cases in "else"?

Reply:
I don't see another place where only `params.field_mask = UCT_MD_MEM_DEREG_FIELD_MEMH;` is initialized

Reply:
ok